docs/operator-manual/applicationset/Generators-Merge.md

```diff
@@ -111,6 +111,68 @@ When merged with the updated base parameters, the `values.redis` value for the p
   values.redis: 'true'
 ```
 
+## Example: Use value interpolation in merge
+
+Some generators support additional values and interpolating from generated variables to selected values. This can be used to teach the merge generator which generated variables to use to combine different generators.
+
+The following example combines discovered clusters and a git repository by cluster labels and the branch name:
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: cluster-git
+spec:
+  generators:
+    # merge 'parent' generator:
+    # Use the selector set by both child generators to combine them.
+    - merge:
+        mergeKeys:
+          - values.selector
```

Comment:
@girstenbrei does this work for go-templated appsets? I think in that case `values` is a proper object. I'm not sure whether this key will behave.

Reply:
Hi @crenshaw-dev , good catch: Just tested this, indeed, with go-template enabled this does not work (the merge key resolves to "values.selector" and not ~"values"."selector". This seems like a current limitation of the merge generator; using go-templating the generator is not able to merge on a nested key.

If that's ok with you, I'd adapt the documentation to reflect this and open another issue regarding the merge operator supporting nested keys, keeping the rest of this PR as is?

Reply:
Makes sense, thanks!